src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcProxyShimFactory.cs

```diff
@@ -30,6 +30,8 @@ internal sealed class DtcProxyShimFactory
     private readonly ConcurrentQueue<ITransactionTransmitter> _cachedTransmitters = new();
     private readonly ConcurrentQueue<ITransactionReceiver> _cachedReceivers = new();
 
+    private static readonly int _maxCachedInterfaces = Environment.ProcessorCount * 2;
```

Comment:
Nit: s_maxCachedInterfaces

A comment as to how this value was chosen would also be helpful.

Reply:
This was copied over as-is from the .NET Framework C++ code, where there wasn't any explanation either. The objects being cached ([ITransactionReceiver](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ms679193(v=vs.85)) and [ITransactionTransmitter](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ms682296(v=vs.85))) are used only for a very short time (a single COM call in fact), and then immediately put back into the queue for recycling; so I think the thinking was to have one available per concurrent usage, plus a margin of *2...

Reply:
Ok